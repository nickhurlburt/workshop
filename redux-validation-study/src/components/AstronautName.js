import React from 'react';

const AstronautName = (props) => {
  return (
    <div>
      <label htmlFor="name">Name</label>
      <input type="text" onBlur={props.onBlur} onChange={(e) => props.updateField(e.target.value)} value={props.field} />
    </div>
  );
};

export default AstronautName;