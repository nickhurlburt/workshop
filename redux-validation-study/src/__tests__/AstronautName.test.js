import React from 'react';
import AstronautName from '../components/AstronautName';
import renderer from 'react-test-renderer';

test('AstronautName renders as planned', () => {
  const onBlur = jest.fn();
  const updateField = jest.fn();
  const component = renderer.create(
    <AstronautName updateField={updateField} onBlur={onBlur} field="name" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});