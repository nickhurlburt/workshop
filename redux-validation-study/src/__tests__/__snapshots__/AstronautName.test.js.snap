// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstronautName renders as planned 1`] = `
<div>
  <label
    htmlFor="name"
  >
    Name
  </label>
  <input
    onBlur={[MockFunction]}
    onChange={[Function]}
    type="text"
    value="name"
  />
</div>
`;
